server/src/proxy/grpc/write.rs

```diff
@@ -576,6 +537,98 @@ fn convert_proto_value_to_datum(
     }
 }
 
+fn find_new_columns(
+    schema: &Schema,
+    schema_config: &SchemaConfig,
+    write_req: &WriteTableRequest,
+) -> Result<Vec<ColumnSchema>> {
+    let new_schema = build_schema_from_write_table_request(schema_config, write_req)
+        .box_err()
+        .context(ErrWithCause {
+            code: StatusCode::INTERNAL_SERVER_ERROR,
+            msg: "Build schema from write table request failed",
+        })?;
+
+    let columns = new_schema.columns();
+    let old_columns = schema.columns();
+    let mut new_columns = Vec::new();
+
+    columns.iter().for_each(|column| {
+        if !old_columns.iter().any(|c| c.name == column.name) {
+            new_columns.push(column.clone());
+        }
+    });
+    Ok(new_columns)
+}
+
+async fn execute_add_columns_plan<Q: QueryExecutor + 'static>(
+    request_id: RequestId,
+    catalog: &str,
+    schema: &str,
+    instance: InstanceRef<Q>,
+    table: TableRef,
+    columns: Vec<ColumnSchema>,
+    deadline: Option<Instant>,
+) -> Result<()> {
+    let table_name = table.name().to_string();
+    info!("Add columns start, request_id:{request_id}, table:{table_name}, columns:{columns:?}");
+
+    let plan = Plan::AlterTable(AlterTablePlan {
+        table,
+        operations: AlterTableOperation::AddColumn(columns),
+    });
+    let _ = execute_plan(request_id, catalog, schema, instance, plan, deadline)
+        .await
+        .map_err(|e| {
+            error!("Add colum failed, err:{e}");
```

Comment:
This error log seems not necessary, you can use with_context to add info you want.